desktop-exporter/trace_store.go

```diff
@@ -32,6 +32,35 @@ func (store *TraceStore) Add(_ context.Context, spanData SpanData) {
 	store.traceMap[spanData.TraceID] = append(store.traceMap[spanData.TraceID], spanData)
 }
 
+func (store *TraceStore) GetRecentTraceIDs(traceCount int) []string {
+	store.mut.Lock()
+	defer store.mut.Unlock()
+
+	if traceCount > store.traceQueue.Len() {
+		traceCount = store.traceQueue.Len()
+	}
+
+	recentTraceIDs := make([]string, 0, traceCount)
+	element := store.traceQueue.Front()
+
+	for i := 0; i < traceCount; i++ {
+		recentTraceIDs = append(recentTraceIDs, element.Value.(string))
+		element = element.Next()
+	}
+
+	return recentTraceIDs
+}
+
+func (store *TraceStore) GetTraceByID(traceID string) ([]SpanData, error) {
```

Comment:
It might be good to create an output type here that we can add more data to rather than an array. Something like:

```go
type struct TraceData {
  Spans []SpanData,
  DurationMS int64,
  RootServiceName string,
  // extend here
}

func (store *TraceStore) GetTraceByID(traceID string) (TraceData, error) {
```

or maybe this should live in the server? 🤔 What do you think?